packages/storybook/src/global-type-extensions.d.ts

```diff
@@ -0,0 +1,28 @@
+// This file has modifications to global types used for nimble-components builds:
```

Comment:
In order to get nimble-components source to successfully build these globals defined in nimble-components needed to get redefined here (they are used only during source build). To me this is more of an indication that this pattern isn't great. I think we should try and enable `isolatedModules` in the tsconfig and then find an alternative pattern to these d.ts files. For now just copied the declaration from nimble to here.

Reply:
Could you add comments at the top of both copies that points at the other copy and indicates they should be kept in sync?

It also looks like we are using a recent enough typescript version (5.1) to remove the `HTMLDialogElement` interface (4.8) but not quite the `NumberFormatOptions` one (5.5). Wanna try removing the dialog one, either in this PR or a follow up?

Reply:
Added a comment and removed the dialog part in this PR

Reply:
Also made the components change type 'none' since it should be a nimble repo build time type checking change and not have an impact on published packages.